test/function.js

```diff
@@ -31,6 +31,23 @@ describe('func', () => {
         ], done);
     });
 
+    it('validates a function arity', (done) => {
+
+        Helper.validate(Joi.func().arity(2).required(), [
+            [function (a,b) { }, true],
```

Comment:
You should also test :
- error messages (eg. https://github.com/hapijs/joi/blob/master/test/any.js#L1463)
- arrow functions